src/Lecture1.hs

```diff
@@ -123,4 +143,17 @@ and lower than 6 elements (4, 5, 6, 7, 8 and 9).
 
 🕯 HINT: Use recursion to implement this function.
 -}
-lowerAndGreater n list = error "TODO"
+
+--No idea how to implement this with the recursion unfortunately.
+
+lowerAndGreater :: Int -> [Int] -> [Char]
+lowerAndGreater n list = show n ++ " is greater than " ++ show (length (filter (<n) list)) ++ " elements and lower than " ++ show (length (filter (>n) list)) ++ " elements"
```

Comment:
That's correct! 💯 

However, with this solution, you will walk through the list twice.
With the recursive solution and manual check on every element in the list on both conditions at the same time (without `filter` function), you would be able to write it more performant with only one traversal of the list 🙂

To implement a recursive solution in a single traversal, you can create a helper local function `go` inside `where` that takes two numbers: how many elements are lower than `n` and how many elements are bigger than `n`. It would look like this:

```haskell
  where
    go :: Int -> Int -> [Int] -> [Char]
    go lower great l = ...
```

Inside this function, you can implement logic similar to the `count` function from slides: you check if the list is empty or not. If it's empty, you return the expected string. Otherwise, you compare the head of the list with `n` and call `go` recursively either increasing one of its arguments or not increasing anything at all.

Reply:
Will try to implement it recursive way, a bit later. Thanks for the suggestions!

Reply:
Finally getting back to this. This should do now :)